sympy/solvers/bivariate.py

```diff
@@ -205,11 +241,33 @@ def _solve_lambert(f, symbol, gens):
     if not lamcheck:
         raise NotImplementedError()
 
+    # replacing all even_degrees of symbol with dummy variable t.
+    # lhs can also be different from Mul and Add which doesn't need
+    # to be processed.
+    t = Dummy('t', **symbol.assumptions0)
+    if lhs.is_Add or lhs.is_Mul:
+        lhs = lhs.replace(lambda i:i.is_Pow and i.base == symbol and \
```

Comment:
How about something like:

```py
        def is_sub_expr_even_power_symbol(sub_expr):
            """Checks if the expression is given symbol with even exponent"""
            has_even_power = sub_expr.exp % 2 == 0
            base_is_symbol = sub_expr.base == symbol
            return sub_expr.is_Pow and base_is_symbol and has_even_power

        def replace_symbol_with_dummy_var(sub_expr):
            """Replaces sub expression with dummy variable `t` with same power as symbol"""
            return t ** sub_expr.exp

        lhs = lhs.replace(is_sub_expr_even_power_symbol, replace_symbol_with_dummy_var)
```

Note: Don't do a verbatim copy of the code, check first!

Reply:
> ```python
> has_even_power = sub_expr.exp % 2 == 0
```
This will fail if `sub_expr` is not a Pow because non Pow does not have `exp` attribute.

Reply:
I think it will be over documented code if we do this. It's already clear from the comments given above that what we are doing and creating routine for this thing exclusively I don't think is good.

Reply:
@smichr Its a suggestion to write something along those lines which is more readable, and no way a tested code, that's why I mentioned to not do a verbatim copy of the code, the improvement could be
`has_even_power = sub_expr.is_Pow and sub_expr.exp % 2 == 0` instead of checking `is_Pow` in the last statement.

> I think it will be over documented code if we do this.

I disagree with this, the above suggestion is not about documenting the code, its about increasing the readability of the code, which is what lacking heavily in old solve and that's why it's a mess and that's the whole point of writing `solveset` at the first place.

> It's already clear from the comments given above 

I have mentioned before that relying on comments to improve the readability of code is not a great idea, because comments can lie, but code cannot, that's why its really very important to improve the code. As solve being just heuristics and not really standard well known algorithms, its very hard for someone to pick it easily and contribute.

So writing a code which only 3-4 people understand is not a great idea, I cannot stress more on the fact that good code is as important as correct logic.

Reply:
Hmm okay, @aktech I will make sure it's more readable now.

Reply:
Can you check it @aktech ?